ksql-engine/src/test/java/io/confluent/ksql/function/udf/math/RoundTest.java

```diff
@@ -0,0 +1,232 @@
+    assertThat(udf.round(new BigDecimal("100.1115"), 3), is(new BigDecimal("100.112")));
+    assertThat(udf.round(new BigDecimal("100.1117"), 3), is(new BigDecimal("100.112")));
+    assertThat(udf.round(new BigDecimal("12345.67"), -1), is(new BigDecimal("1.235E4")));
+    assertThat(udf.round(new BigDecimal("12345.67"), -2), is(new BigDecimal("1.23E4")));
+    assertThat(udf.round(new BigDecimal("12345.67"), -3), is(new BigDecimal("1.2E4")));
+    assertThat(udf.round(new BigDecimal("12345.67"), -4), is(new BigDecimal("1E4")));
+    assertThat(udf.round(new BigDecimal("12345.67"), -5), is(new BigDecimal("0E5")));
+  }
+
+  @Test
+  public void testRoundBigDecimalWithDecimalPlacesNegative() {
+    assertThat(udf.round(new BigDecimal("-1.0"), 0), is(new BigDecimal("-1")));
+    assertThat(udf.round(new BigDecimal("-1.1"), 0), is(new BigDecimal("-1")));
+    assertThat(udf.round(new BigDecimal("-1.5"), 0), is(new BigDecimal("-1")));
+    assertThat(udf.round(new BigDecimal("-1.75"), 0), is(new BigDecimal("-2")));
+    assertThat(udf.round(new BigDecimal("-100.1"), 0), is(new BigDecimal("-100")));
+    assertThat(udf.round(new BigDecimal("-100.5"), 0), is(new BigDecimal("-100")));
+    assertThat(udf.round(new BigDecimal("-100.75"), 0), is(new BigDecimal("-101")));
+    assertThat(udf.round(new BigDecimal("-100.10"), 1), is(new BigDecimal("-100.1")));
+    assertThat(udf.round(new BigDecimal("-100.11"), 1), is(new BigDecimal("-100.1")));
+    assertThat(udf.round(new BigDecimal("-100.15"), 1), is(new BigDecimal("-100.1")));
+    assertThat(udf.round(new BigDecimal("-100.17"), 1), is(new BigDecimal("-100.2")));
+    assertThat(udf.round(new BigDecimal("-100.110"), 2), is(new BigDecimal("-100.11")));
+    assertThat(udf.round(new BigDecimal("-100.111"), 2), is(new BigDecimal("-100.11")));
+    assertThat(udf.round(new BigDecimal("-100.115"), 2), is(new BigDecimal("-100.11")));
+    assertThat(udf.round(new BigDecimal("-100.117"), 2), is(new BigDecimal("-100.12")));
+    assertThat(udf.round(new BigDecimal("-100.1110"), 3), is(new BigDecimal("-100.111")));
+    assertThat(udf.round(new BigDecimal("-100.1111"), 3), is(new BigDecimal("-100.111")));
+    assertThat(udf.round(new BigDecimal("-100.1115"), 3), is(new BigDecimal("-100.111")));
+    assertThat(udf.round(new BigDecimal("-100.1117"), 3), is(new BigDecimal("-100.112")));
+    assertThat(udf.round(new BigDecimal("-12345.67"), -1), is(new BigDecimal("-1.235E4")));
+    assertThat(udf.round(new BigDecimal("-12345.67"), -2), is(new BigDecimal("-1.23E4")));
+    assertThat(udf.round(new BigDecimal("-12345.67"), -3), is(new BigDecimal("-1.2E4")));
+    assertThat(udf.round(new BigDecimal("-12345.67"), -4), is(new BigDecimal("-1E4")));
+    assertThat(udf.round(new BigDecimal("-12345.67"), -5), is(new BigDecimal("-0E5")));
+  }
+
+  /*
+  These test some numbers which can't be represented exactly as doubles and end with 5
```

Comment:
Try to replace comments with appropriate test names, e.g. `shoudlHandleDecimalsThatCanNotBeRepresentedByDouble`